test/test-lex.cpp

```diff
@@ -113,6 +113,114 @@ TEST_F(test_lex, lex_block_comments) {
   }
 }
 
+TEST_F(test_lex, lex_unopened_block_comment) {
+  {
+    error_collector v;
+    padded_string input(u8"hello */"_sv);
+    lexer l(&input, &v);  // identifier
+    EXPECT_EQ(l.peek().type, token_type::identifier);
+    l.skip();  // end of file
+    EXPECT_EQ(l.peek().type, token_type::end_of_file);
+    EXPECT_THAT(v.errors,
+                ElementsAre(ERROR_TYPE_FIELD(
+                    error_unopened_block_comment, comment_close,
+                    offsets_matcher(&input, strlen(u8"hello "), u8"*/"))));
+  }
+  {
+    error_collector v;
+    padded_string input(u8"*-----*/"_sv);
+    lexer l(&input, &v);
+
+    while (l.peek().type != token_type::end_of_file) {
+      l.skip();
+    }
+    EXPECT_EQ(l.peek().type, token_type::end_of_file);
+
+    EXPECT_THAT(v.errors,
+                ElementsAre(ERROR_TYPE_FIELD(
+                    error_unopened_block_comment, comment_close,
+                    offsets_matcher(&input, strlen(u8"*-----"), u8"*/"))));
+  }
+  {
+    error_collector v;
+    padded_string input(u8"*******/"_sv);
+    lexer l(&input, &v);
+    EXPECT_EQ(l.peek().type, token_type::star_star);
+    l.skip();
+    EXPECT_EQ(l.peek().type, token_type::star_star);
+    l.skip();
+    EXPECT_EQ(l.peek().type, token_type::star_star);
+    l.skip();
+    EXPECT_EQ(l.peek().type, token_type::end_of_file);
+
+    EXPECT_THAT(v.errors,
+                ElementsAre(ERROR_TYPE_FIELD(
+                    error_unopened_block_comment, comment_close,
+                    offsets_matcher(&input, strlen(u8"******"), u8"*/"))));
+  }
+  {
+    error_collector v;
+    padded_string input(u8"*/"_sv);
+    lexer l(&input, &v);
+    EXPECT_EQ(l.peek().type, token_type::end_of_file);
+
+    EXPECT_THAT(v.errors, ElementsAre(ERROR_TYPE_FIELD(
+                              error_unopened_block_comment, comment_close,
+                              offsets_matcher(&input, 0, u8"*/"))));
+  }
+  {
+    error_collector v;
+    padded_string input(u8"**/"_sv);
+    lexer l(&input, &v);
+    EXPECT_EQ(l.peek().type, token_type::end_of_file);
+    EXPECT_THAT(v.errors, ElementsAre(ERROR_TYPE_FIELD(
+                              error_unopened_block_comment, comment_close,
+                              offsets_matcher(&input, 1, u8"*/"))));
+  }
+}
+
+TEST_F(test_lex, lex_regexp_literal_starting_with_star_slash) {
+  {
+    error_collector v;
+    padded_string input(u8"3; */ hello/"_sv);
+    lexer l(&input, &v);
+    EXPECT_EQ(l.peek().type, token_type::number);
+    l.skip();
+    EXPECT_EQ(l.peek().type, token_type::semicolon);
+    l.skip();
```

Comment:
Nit: I'd omit `3;` because it's irrelevant to the test.
```suggestion
    padded_string input(u8"*/ hello/"_sv);
    lexer l(&input, &v);
```